src/homework/homeworks/Animal.java

```diff
@@ -0,0 +1,84 @@
+package homework.homeworks;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class Animal implements Serializable {
+    private final String name;
+
+    public Animal(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof Animal) {
+            return Objects.equals(name, ((Animal) obj).name);
+        }
+        return false;
+    }
+
+    @Override
+    public String toString() {
+        return "Animal{" +
+                "name='" + name + '\'' +
+                '}';
+    }
+
+    public static Animal[] deserializeAnimalArray(byte[] data) {
+        try (ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(data);
+             ObjectInputStream inputReader = new ObjectInputStream(byteArrayInputStream);) {
+
+            int sizeOfArray = inputReader.readInt();
+            Animal[] animals = new Animal[sizeOfArray];
+            for (int i = 0; i < sizeOfArray; i++) {
+                Object obj = inputReader.readObject();
+                if (obj instanceof Animal) {
+                    animals[i] = (Animal) obj;
+                } else {
+                    throw new IllegalArgumentException();
+                }
+            }
+            return animals;
+        } catch (IOException | ClassNotFoundException e) {
```

Comment:
e is not valid name

Reply:
As I know if often used in java and saw a lot of examples. Is not correct to leave it like this? What is should be?